integration-tests/java-joor-dsl/src/main/java/org/apache/camel/quarkus/dsl/java/joor/JavaJoorDslResource.java

```diff
@@ -29,10 +31,12 @@
 import org.apache.camel.ExtendedCamelContext;
 import org.apache.camel.ProducerTemplate;
 import org.apache.camel.Route;
+import org.apache.camel.component.direct.DirectEndpoint;
 import org.apache.camel.dsl.java.joor.JavaRoutesBuilderLoader;
 import org.apache.camel.quarkus.main.CamelMain;
 import org.apache.camel.spi.RoutesBuilderLoader;
 
+@RegisterForReflection(targets = String.class)
```

Comment:
Just curious - what requires reflective access to `String`?

Reply:
Good question/remark, it is needed to be able to call [`String#endsWith(String)`](https://github.com/apache/camel-quarkus/pull/4737/files#diff-97f7f2c0a7818c93c92fda67bdee1c0c759db70dd89dee26693183ca59705988R38) in a simple expression